JASP-Engine/JASP/R/commonerrorcheck.R

```diff
@@ -552,3 +553,39 @@
   }
   return(result)
 }
+
+.checkVarCovData <- function(dataset, target, corFun = NULL, grouping=NULL, corArgs = NULL, ...) {
+  # Check if the matrix returned by corFun is positive definite. 
+  # internally uses .checkVarCovMatrix
+  # Args:
+  #   dataset: JASP dataset.
+  #   target: String vector indicating the target variables.
+  #   corFun: a function that calculates a correlation matrix or covariance matrix (e.g. cor or cov are recommended)
+  #   corArgs: more arguments to corFun, i.e. use = "pairwise".
+  #   grouping: String vector indicating the grouping variables.
+  
+  result <- list(error=FALSE, errorVars=NULL)
+  
+  if (is.null(corFun))
+    corFun <- stats::cor
+
+  if (is.null(grouping)) {
+    dataset <- list(dataset)
+  } else {
+    groupingData <- dataset[[.v(grouping)]]
```

Comment:
Would .applyOnGroups() work for this?

Reply:
I tried some things, but `.applyOnGroups` doesn't work for me. The problem is with the line `func(data[[.v(target)]])`. In my case, `data[[.v(target)]]` must return something that can be interpreted as a matrix. However, due to the double `[[` target cannot have length larger than 1. We could make an `.applyOnGroups` that allows for matrix returns, but so far this is the only use case.